### DISCUSSION TITLE
"Green CI"

### DISCUSSION THREAD

**Discussion by metazool** [ukceh-rse/community#18] [General]
I heard some discussion of "Green CI" at a recent RSE-adjacent event - folks talking about optimising their continuous integration practises to save energy costs.

I'm finding the topic hard to search for, though! I'd be interested in seeing a compilation of references and tips for small things to change, especially with specific reference to Github Actions

Trying to be more mindful about simple things that some people probably do as a matter of course - setting `paths` on pipelines so that documentation changes don't trigger test runs on code that hasn't changed; re-ordering steps so that linting runs first before installing complex dependencies needed for testing.

The default Ubuntu base images are probably heavier than they need to be, would there be debian or alpine based images more suited for python testing? I thought this was where [distroless](https://medium.com/@bubu.tripathy/power-of-distroless-images-945cceb4b33e) containers were meant to shine, but can't see Github Actions reference points for that.... 

Could also be a topic for a "Monday club" session?

**Comment by metazool** [comment]
@dolegi tagging you here in case this is a topic you have Opinions about...

**Comment by dolegi** [comment]
This is an interesting angle to look at it from, I'm used to focussing on making CI quick, which could have some parallels with "Green".

We hit the artifact storage limit with our CI/CD pipeline because of the github plan we are using and had to do to some reworking there (@lewis-chambers did the fix and might be able to eloborate), not sure if the fix would increase or decrease energy costs, not something we considered at the time.

I hadn't seen distroless containers before sounds interesting and more practical than [unikernels](http://unikernel.org/) but I guess unikernels probably use less energy. You might already be aware but there's a unikernel company that has links to lancaster university https://unikraft.org/
